src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -423,18 +430,43 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     }
 
     fun addSuperinterfaces(superinterfaces: Iterable<TypeName>) = apply {
-      this.superinterfaces += superinterfaces
+      this.superinterfaces.putAll(superinterfaces.map {
```

Comment:
off-topic: this seems like an inefficient idiom. I think it’s fine for this project but it makes me anxious because it creates a lot of temporary stuff for syntactic benefit